app/util/ReferrerUrlChecking.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package util
+
+import config.{AppConfig, CFConfig}
+import play.api.Logging
+import play.api.mvc.{AnyContent, Request}
+import uk.gov.hmrc.play.bootstrap.binders.RedirectUrl._
+import uk.gov.hmrc.play.bootstrap.binders._
+
+trait ReferrerUrlChecking extends Logging {
+
+  implicit class OptionStringOps(opt: Option[String]) {
+    private val notApplicable               = "n/a"
+    def orNotApplicable: Option[String]     = Some(opt.getOrElse(notApplicable))
+    def removeNotApplicable: Option[String] = opt match {
+      case Some(referrer) if referrer == notApplicable => None
+      case _                                           => opt
+    }
+  }
+
+  def logInvalidUrl(service: Option[String], url: RedirectUrl)(implicit appConfig: AppConfig): SafeRedirectUrl =
+    url.getEither(appConfig.urlPolicy) match {
+      case Right(safeRedirectUrl: SafeRedirectUrl) => safeRedirectUrl
+      case Left(unsafeRedirectError)               =>
+        val serviceName = if (service.isEmpty) "A service" else s"Service [${service.get}]"
+        logger.warn(s"$serviceName is attempting to redirect to an unsafe url: $unsafeRedirectError")
```

Comment:
I'd maybe match the log format of https://github.com/hmrc/accessibility-statement-frontend/pull/1119/files#diff-ec5bcdd149a3a2cdc3dd99f287988f3271e7246873fc307e662ef7c24e8c9fcaR64 here - assuming unsafeRedirectError has enough context in the error